StreamChatClient_v3/Workers/EventObservers/EventObserver_Tests.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient_v3
+import XCTest
+
+final class EventObserver_Tests: XCTestCase {
+    var notificationCenter: NotificationCenter!
+    var eventToDeliver: HealthCheckEvent!
+    var eventNotification: Notification {
+        .init(newEventReceived: eventToDeliver, sender: self)
+    }
+
+    // MARK: - Setup
+
+    override func setUp() {
+        notificationCenter = NotificationCenter()
+        eventToDeliver = HealthCheckEvent(connectionId: .unique)
+    }
+
+    override func tearDown() {
+        notificationCenter = nil
+        eventToDeliver = nil
+    }
+
+    // MARK: - Calling back tests
+
+    func test_callbackIsNotCalled_ifObserverIsDeallocated() {
+        var callbackExecutionCount = 0
+
+        // Create observer and count callback executions
+        var observer: EventObserver? = .init(notificationCenter: notificationCenter,
+                                             tranform: { $0 as? HealthCheckEvent },
+                                             callback: { _ in callbackExecutionCount += 1 })
+        _ = observer
+
+        // Send event notification
+        notificationCenter.post(eventNotification)
+        // Assert callback is called
+        XCTAssertEqual(callbackExecutionCount, 1)
```

Comment:
I'd suggest making these assertions async since the notification callback doesn't have to happen synchronously. `AssertAsync.willBeEqual` here ...

Reply:
fixed in c79a9c3ebd4f36f3dbb1bcc63f4b4a544765467a